tests/internal.spec.js

```diff
@@ -0,0 +1,198 @@
+var ncl = require('../lib/login.js'),
+    nclWrapper = require('../lib'),
+    expect = require('chai').expect,
+    path = require('path')
+
+describe('Can handle', function () {
+  it('missing username', function () {
+    expect(function () {
+      nclWrapper();
+    }).to.throw('AssertionError');
+  });
+
+  it('missing password', function () {
+    expect(function () {
+      nclWrapper('username');
+    }).to.throw('AssertionError');
+  });
+
+  it('missing email', function () {
+    expect(function () {
+      nclWrapper('username', 'password');
+    }).to.throw('AssertionError');
+  });
+});
+
+describe('Can resolve', function () {
+  it('custom username', function () {
+    var args = ncl.processArguments('username', 'password', 'test@test.com');
+    expect(args).to.have.property('user', 'username');
+  });
+
+  it('custom password', function () {
+    var args = ncl.processArguments('username', 'password', 'test@test.com');
+    expect(args).to.have.property('pass', 'password');
+  });
+
+  it('custom email', function () {
+    var args = ncl.processArguments('username', 'password', 'test@test.com');
+    expect(args).to.have.property('email', 'test@test.com');
+  });
+
+  it('default registry', function () {
+    var args = ncl.processArguments('username', 'password', 'test@test.com');
```

Comment:
All the string constants here should be collected into string variables, so it is easier to change.